File: task-03/prime.c
#include <stdio.h>
#include <stdbool.h>

bool isPrime(int num)
{
    if (num <= 1)
        return false;

    for (int i = 2; i < num; i++)
        if (num % i == 0)
            return false;

    return true;
}

int main()
{
    int n;
    printf("Enter any number: ");
    scanf("%i", &n);

    for (int i = 2; i <= n; i++)
        if (isPrime(i))
            printf("%d ", i);

    return 0;
}

File: task-03/prime.hs
isPrime :: Int -> Bool
isPrime num
    | num <= 1   = False
    | otherwise  = all (\x -> num `mod` x /= 0) [2..num-1]

findPrimes :: Int -> [Int]
findPrimes n = [x | x <- [2..n], isPrime x]

main :: IO ()
main = do
    putStrLn "Enter any number: "
    input <- getLine
    let n = read input :: Int
    let primes = findPrimes n
    putStrLn (show primes)
